src/lib/storage/dictionary_column.hpp

```diff
@@ -1,87 +1,66 @@
 #pragma once
 
 #include <memory>
-#include <optional>
 #include <string>
-#include <utility>
-#include <vector>
 
-#include "all_type_variant.hpp"
 #include "base_dictionary_column.hpp"
 #include "types.hpp"
 
 namespace opossum {
 
-class BaseAttributeVector;
-class BaseColumn;
+class BaseZeroSuppressionVector;
```

Comment:
Why don’t we keep this name? It’s very precise. Zero suppression means that a vector is compressed and that this vector is an unsigned integer. It also explains how the vector is compressed, by suppressing zeros.

Reply:
"Zero suppression" is not a term that people necessarily associate with using only as few bytes as necessary. Taking two links from the Google Search Results for "zero suppression", I find three concepts that share that name and none of it is what we are doing:

https://en.wikipedia.org/wiki/Zero_suppression
https://www.ibm.com/support/knowledgecenter/en/SSY2VQ_2.0.0/com.ibm.aix.pli.doc/zrchar.htm

Also, It lacks the info that it holds only integers.

Reply:
> I find three concepts that share that name and none of it is what we are doing.

Yes, because the actual concept is called null suppression, but someone asked me to rename it.

Reply:
I’m sorry. I’m just hesitant because it would mean having to rename a lot more then just the vector, every class or enum or method containing `zs`, etc. I’d probably go for `BaseCompressedVector`.

Reply:
Null Suppression suffers from the same problem:
http://sigmodrecord.org/publications/sigmodRecord/9309/compress.ps (RLE for NULL values)
https://docs.oracle.com/en/cloud/paas/bi-cloud/bilug/suppressing-null-values-views.html (Hiding NULL values in the frontend)
https://books.google.de/books?id=A5UQY64GCdcC&ots=La4ayU2CJm&dq=%22null%20suppression%22&lr&hl=de&pg=PA3#v=onepage&q=%22null%20suppression%22&f=false (Null Suppression and Zero Suppression used interchangeably, again about NULL values, but different).

I know that those naming discussions and the actual renaming are both a giant PITA, sorry. :/